platform/pulp/platform/models/progress.py

```diff
@@ -0,0 +1,195 @@
+"""
+Django models related to Progress Reporting
+"""
+from gettext import gettext as _
+import logging
+
+from django.contrib.contenttypes import fields
+from django.db import models
+
+from pulp.platform.models import Model, Task
+
+
+__all__ = [
+    'ProgressBar',
+    'ProgressReportingBase',
+    'ProgressSpinner',
+]
+
+
+_logger = logging.getLogger(__name__)
+
+
+class ProgressReportingBase(Model):
+    """
+    An abstract base model for all progress reporting.
+
+    All progress reports have a message, state, and are related to a Task.
+
+    Fields:
+
+    :cvar message: A short message for the progress update, typically shown to the user. (required)
+    :type message: models.TextField
+
+    :cvar state: The state of the progress update. Defaults to `WAITING`. This field uses a limited
+                 set of choices of field states. See `STATES` for possible states.
+    :type state: models.TextField
+
+    Relations:
+
+    :cvar task: The task associated with this progress report. If left unset when save() is called
+                it will be set to the current task_id.
+    :type task: models.ForeignKey
+    """
+    WAITING = 'waiting'
+    SKIPPED = 'skipped'
+    RUNNING = 'running'
+    FINISHED = 'finished'
+    ERROR = 'error'
+    CANCELED = 'canceled'
+    STATES = (
+        (WAITING, 'Waiting'),
+        (SKIPPED, 'Skipped'),
+        (RUNNING, 'Running'),
+        (FINISHED, 'Finished'),
+        (ERROR, 'Error'),
```

Comment:
This is in present tense, but it likely should be in past tense since it's a terminal state.
s/Error/Failed/